main.js

```diff
@@ -210,6 +274,79 @@ function animate(renderer, labelRenderer, camera, controls) {
     labelRenderer.render(scene, camera);
 }
 
+function clearGameBoard(scene, gameBoard) {
+    for (let stone of stonesInScene) {
+        scene.remove(stone);
+    }
+    stonesInScene = [];
+
+    for (let i = 0; i < gameBoard.length; i++) {
+        for (let j = 0; j < gameBoard[i].length; j++) {
+            if (gameBoard[i][j] !== null) {
+                gameBoard[i][j] = null;
+            }
+        }
+    }
+}
+
+function restartGame(gameBoard) {
+    return function (event) {
+        event.preventDefault();
+        clearGameBoard(scene, gameBoard);
+        currentPlayer = 'black';
+        currentPlayerLabel.element.textContent = `Player turn: ${currentPlayer}`;
+
+        if (winnerAnnouncement) {
+            scene.remove(winnerAnnouncement);
+            winnerAnnouncement = null;
+        }
+    }
+}
```

Comment:
It looks like this is a factory function. That is, it creates an event handler. If that's the intention, it's typical to include that in the name. Maybe `restartGameHandlerFactory`